File: term4/database-applications/project/Store/Store.Infrastructure/Repositories/CustomerRepository/TestCustomerRepository.cs
using Store.Common;
using Store.Domain.Model.Customer;
using Store.Domain.Model.Shared;
using System.Collections.Generic;

namespace Store.Infrastructure.Repositories.CustomerRepository
{
    public class TestCustomerRepository : GenericTestRepository<Customer>, ICustomerRepository
    {
        public TestCustomerRepository() : this(new List<Customer> {
            new Customer { Id = 1, Name = "test", Email = "test@test.com",
                Address = new Address { Country = "test", City = "test", Detail = "test", ZipCode = "12-345"}  },
            new Customer { Id = 2, Name = "test", Email = "test@test.com",
                Address = new Address { Country = "test", City = "test", Detail = "test", ZipCode = "12-345"}  },
            new Customer { Id = 3, Name = "test", Email = "test@test.com",
                Address = new Address { Country = "test", City = "test", Detail = "test", ZipCode = "12-345"}  },
        }) { }

        private TestCustomerRepository(ICollection<Customer> customers) : base(customers) { }
    }
}

File: term4/database-applications/project-old/src/main/kotlin/dbapp/project/infrastructure/test/TestCustomerRepository.kt
package dbapp.project.infrastructure.test

import dbapp.project.common.GenericTestRepository
import dbapp.project.domain.customer.Customer
import dbapp.project.domain.customer.CustomerRepository
import dbapp.project.domain.shared.Address

class TestCustomerRepository : GenericTestRepository<Customer>(
        mutableSetOf(
                Customer(1, "Test1", "Test1", "test1@test.com", null, null),
                Customer(2, "Test2", "Test2", "test2@test.com", "123123123", null),
                Customer(3, "Test3", "Test3", "test3@test.com", null,
                        Address("Test", "Test", "Test 21/37", "12-345"))
        )
), CustomerRepository